crypto/s2n_ecc.c

```diff
@@ -31,9 +31,12 @@
 /* IANA values can be found here: https://tools.ietf.org/html/rfc8446#appendix-B.3.1.4 */
 /* Share sizes are described here: https://tools.ietf.org/html/rfc8446#section-4.2.8.2
  * and include the extra "legacy_form" byte */
+
 const struct s2n_ecc_named_curve s2n_ecc_supported_curves[2] = {
-    {.iana_id = TLS_EC_CURVE_SECP_256_R1, .libcrypto_nid = NID_X9_62_prime256v1, .name = "secp256r1", .share_size = ( 32 * 2 ) + 1 },
-    {.iana_id = TLS_EC_CURVE_SECP_384_R1, .libcrypto_nid = NID_secp384r1, .name= "secp384r1", .share_size = ( 48 * 2 ) + 1 },
+    [S2N_ECC_NAMED_CURVED_SECP_256_R1] = {.iana_id = TLS_EC_CURVE_SECP_256_R1, .libcrypto_nid = NID_X9_62_prime256v1, .name = "secp256r1",
```

Comment:
I'm not sure I like how this table is handled. We create an enum of 0,1,... elements. Then we create an array where we explicitly set the 0,1,... elements, and also include the same 0,1,... labels in the elements themselves. We're recording this ordering in 3 different ways. Imagine what would happen if we made a typo somewhere.

Is there any way to avoid that? Could we have one source of truth for the ordering, and have an init function build all other versions of the ordering from that single ordering? Is it even important that we get constant time lookups on a table this small?